src/Onboarding/Routes/SettingsRoute.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Give\Onboarding\Routes;
+
+use WP_REST_Request;
+use Give\API\RestRoute;
+
+class SettingsRoute implements RestRoute {
+
+	protected $endpoint = 'onboarding/settings';
+
+	public function handleRequest( WP_REST_Request $request ) {
+		return [
+			'data' => [
+				'setting' => $request->get_param( 'setting' ),
+				'value'   => json_decode( $request->get_param( 'value' ) ),
+			],
+		];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'POST',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => function() {
+						return current_user_can( 'manage_options' );
+					},
+					'args'                => [
+						'setting' => [
+							'type'              => 'string',
+							'required'          => true,
+							// 'validate_callback' => [ $this, 'validateSetting' ],
```

Comment:
I think it's fine that the validation callback for this endpoint is unfinished in this PR, since it is pretty reliant on exactly which settings we ultimately decide a user can persist. Just wanting to point this out to make sure it gets picked up in your Onboarding Wizard setting persistence issue. 👍

Reply:
Cool. I referenced this note in the acceptance criteria for https://github.com/impress-org/givewp/issues/4979